### PR TITLE
Feature/blue 1357 orbit resource crud

### PR DESCRIPTION
**Pull request by coutinol**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@ HOSTNAME=registry.terraform.io
 NAMESPACE=leanspace
 NAME=leanspace
 BINARY=terraform-provider-${NAME}
-VERSION?=0.4.0
```
Comment (jerome-leanspace) [inline Makefile:6]: either let it to the previous value or put a really high version like `100.0.0`

File: provider/client.go [hunk 2]
```diff
@@ -40,10 +40,10 @@ func NewClient(host, env, tenant, clientId, clientSecret *string) (*Client, erro
 		env = &environment
 	}
 	hostUrl := "https://api.leanspace.io"
-	switch *env {
-	case "develop", "demo":
-		hostUrl = fmt.Sprintf("https://api.%s.leanspace.io", *env)
-	}
+    switch *env {
```
Comment (jerome-leanspace) [inline provider/client.go:43]: why this change?

File: services/orbits/orbit_resources/parsers.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+func (orbitResource *OrbitResource) ToMap() map[string]any {
+	orbitResourceMap := make(map[string]any)
+	orbitResourceMap["id"] = orbitResource.ID
+	orbitResourceMap["satellite_id"] = orbitResource.SatelliteId
+	orbitResourceMap["name"] = orbitResource.Name
+	orbitResourceMap["data_source"] = orbitResource.DataSource
+	orbitResourceMap["automatic_tle_update"] = orbitResource.AutomaticTleUpdate
+	orbitResourceMap["automatic_propagation"] = orbitResource.AutomaticPropagation
+    orbitResourceMap["gps_metric_ids"] = []any{orbitResource.GpsMetricIds.ToMap()}
```
Comment (jerome-leanspace) [inline services/orbits/orbit_resources/parsers.go:15]: padding

File: services/orbits/orbit_resources/schemas.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+		Type:         schema.TypeString,
+		Required:     true,
+		ForceNew:     true,
+		ValidateFunc: validation.IsUUID,
+	},
+	"name": {
+		Type:     schema.TypeString,
+		Required: true,
+	},
+    "data_source": {
```
Comment (jerome-leanspace) [inline services/orbits/orbit_resources/schemas.go:25]: the padding is weird

File: services/orbits/orbit_resources/schemas.go [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        Optional:    true,
+    },
+    "automatic_propagation": {
+        Type:     schema.TypeBool,
+        Optional:    true,
+    },
+    "gps_metric_ids": {
+        Type:     schema.TypeList,
+        Optional:    true,
+        MinItems: 0,
```
Comment (jerome-leanspace) [inline services/orbits/orbit_resources/schemas.go:40]: MinItems: 1 just for consistency

File: services/orbits/orbit_resources/schemas.go [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Who modified it the last",
+	},
+}
+
+var gpsMetricIdsSchema = map[string]*schema.Schema{
+	"metric_id_for_position_x": {
+		Type:     schema.TypeString,
+		Optional: true,
```
Comment (jerome-leanspace) [inline services/orbits/orbit_resources/schemas.go:71]: aren't they all required when the gpsMetricIds is provided?

File: testing/orbits/orbit_resources/main.tf [hunk 7]
```diff
@@ -0,0 +1,72 @@
+terraform {
+  required_providers {
+    leanspace = {
+      source = "leanspace/leanspace"
+      version = "3.1.0"
```
Comment (jerome-leanspace) [inline testing/orbits/orbit_resources/main.tf:5]: can you remove the version?

File: testing/orbits/orbit_resources/main.tf [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  }
+}
+
+provider "leanspace" {
```
Comment (jerome-leanspace) [inline testing/orbits/orbit_resources/main.tf:10]: can you remove this as well?

**Review by jerome-leanspace — COMMENTED** [review]
I've left a few comments
I'll check if there isn't a better way for the optional gps metric ids because this look complex

File: provider/client.go (hunk 2 above)
Comment (coutinol) [reply provider/client.go:43]: there is something wrong with my padding, let me check

File: services/orbits/orbit_resources/schemas.go (hunk 6 above)
Comment (coutinol) [reply services/orbits/orbit_resources/schemas.go:71]: let me check this, I have tested so many small changes to make the nominal solution work that this one might be useless

**Review by jerome-leanspace — APPROVED** [review]

**Review by VivienLG — APPROVED** [review]
